src/util/amount.rs

```diff
@@ -933,23 +954,46 @@ pub mod serde {
             //! Serialize and deserialize [Optoin<Amount>] as real numbers denominated in satoshi.
```

Comment:
Not related to this PR, but a typo nearby
```suggestion
            //! Serialize and deserialize [Option<Amount>] as real numbers denominated in satoshi.
```

Reply:
Or even better
```suggestion
            //! Serialize and deserialize `Option<`[`Amount`]`>` as real numbers denominated in satoshi.
```